src/middlewares/auth.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const authentication = (req, res, next) => {
+  const bearerHeader = req.headers.authorization;
+  if (bearerHeader) {
+    const bearer = bearerHeader.split(' ');
+    const token = bearer[1];
+    jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
+      if (err) {
+        return res.status(401).json({
+          status: 401,
+          error: 'Unauthorized! token is invalid, please sign in again'
+        });
+      }
+      req.user = decoded;
```

Comment:
Why did are you assigning the decoded payload in `req.user`? I feel `req.decoded` or `req.decodedPayload` is more descriptive.